tests/snippets/testutils.py

```diff
@@ -20,19 +20,27 @@ def assert_raises(exc_type, expr, msg=None):
         assert False, failmsg
 
 
+class Raised:
+    def __init__(self):
+        self.exception = None
+
+
 class assertRaises:
     def __init__(self, expected):
         self.expected = expected
+        self.raised = Raised()
 
     def __enter__(self):
-        return self
+        return self.raised
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         if exc_type is None:
             failmsg = '{} was not raised'.format(self.expected.__name__)
             assert False, failmsg
         if not issubclass(exc_type, self.expected):
             return False
+
+        self.raised.exception = exc_val
```

Comment:
Why do we need the `Raised` class? You can just do:
```py
self.raised_exception = exc_val
```

Reply:
Yeah, you're right.